src/services/Attendance.ts

```diff
@@ -0,0 +1,264 @@
+import StudentDao from "@daos/Airtable/StudentDao";
+import CanvasCoursesDao from "@daos/Airtable/CanvasCoursesDao";
+import ModulesDao from "@daos/Canvas/ModulesDao";
+import SubmissionDao from "@daos/Canvas/SubmissionDao";
+import { FieldSet, Records } from "airtable";
+import Module from "@entities/Module";
+import ModuleItem from "@entities/ModuleItem";
+
+const studentDao = new StudentDao();
+const canvasCoursesDao = new CanvasCoursesDao();
+const modulesDao = new ModulesDao();
+const submissionDao = new SubmissionDao();
+
+interface IMilestone {
+  name: string;
+  itemId: number;
+  points: number;
+  completed: boolean;
+}
+
+/**
+ * Get a set of learners' student records from Airtable by their email.
+ *
+ * @param learners
+ * @returns
+ */
+async function getStudentRecords(
```

Comment:
I think that organizing our service layer like this will be awesome. It keeps things clean and organized and allows for separation of concerns without going over the top when it comes to abstraction.

Reply:
This works. Let's watch as these grow to make sure we become aware of domain boundaries and not fall into pitfalls of the manager patterns.